Source/ACE/Entity/Player.cs

```diff
@@ -710,6 +701,102 @@ public void SpendXp(Skill skill, uint amount)
             Session.Network.EnqueueSend(xpUpdate, skillUpdate, soundEvent, message);
         }
 
+        public void Kill(ObjectGuid killerId)

```

Comment:
I think I'm going to be a really big fan of overriding the existing Kill function (renaming it to OnKill).  Then, here in Player.cs, you'd just override, call base.OnKill(...), then do player respawning stuff.  You could also update the CorpseFactory to pull items off the character, etc, that would drop on death.